design/ceph/ceph-mon-pv.md

```diff
@@ -136,3 +136,132 @@ relies on determining if monitor pods may be placed on a node, requiring
 the k8s schedule thinks otherwise.
+
+## Monitor Storage Class Support
+
+It is generally desirable in production to have multiple zones availability. A
+whole storage class could go unavailable then and enough mons could stay up to
+keep the cluster running.
+In such a case the distribution of mons can be made more homogeneous for each
+zone, by this we can leverage the spread of mons for providing high
+availability. This could be a useful scenario for a datacenter with own storage
+class in contrast to ones provided by cloud providers.
+
+Rook will enable the mon deployment when requested by the admin through the
+usual Rook CRDs.
+
+To support this scenario, we can do something similar to the mon deployment for
+[Stretch clusters](https://github.com/ceph/ceph/blob/master/doc/rados/operations/stretch-mode.rst).
+The operator will have zonal awareness, associating mon deployments based on
+zones configuration provided.
+
+The type of failure domain used for this scenario is commonly "zone", but can
+be set to a different failure domain.
+
+### Failure Domain Configuration
+
+The topology of the K8s cluster is to be determined by the admin, outside the scope of Rook.
+Rook will simply detect the topology labels that have been added to the nodes.
+
+If the desired failure domain is a "zone", the `topology.kubernetes.io/zone` label should
+be added to the nodes. Any of the [topology labels](https://rook.io/docs/rook/latest/ceph-cluster-crd.html#osd-topology)
+supported by OSDs can be used.
+
+For example:
+
+```yaml
+topology.kubernetes.io/zone=a
+topology.kubernetes.io/zone=b
+topology.kubernetes.io/zone=c
+```
+
+###  Design
+
+The core changes to rook are to associate the mons with the required zones.
+
+- Each zone will have at least one mon assigned using labels for nodeAffinity.
+
+The new configuration will support the `zones` configuration where the
+failure domains available must be listed.
+
+```yaml
+  mon:
+    count: 3
+    allowMultiplePerNode: false
+    zones:
+      - name: us-east-2a
+      - name: us-east-2b
+      - name: us-east-2c
+    volumeClaimTemplate:
+      spec:
+        storageClassName: zone-a-storage
+        resources:
+          requests:
+            storage: 10Gi
+```
+
+The operator will track which mons belong to which zone. The zone assignment for each mon
+will be stored in the `rook-ceph-mon-endpoints` configmap in the same data structure where the host
+assignment is stored for mons with node affinity.
+
+For example, if the zones are called
+"zone-a", "zone-b", and "zone-c", the configmap would contain:
+
+```
+data:
+  data: a=10.99.109.200:6789,b=10.98.18.147:6789,c=10.96.86.248:6789,d=10.96.86.249:6789,e=10.96.86.250:6789
+  mapping: '{"node":{"a":"zone-a","b":"zone-b","c":"zone-c"}}'
+  maxMonId: "3"
+```
+
+If no zones are listed in the `spec.mon.zones`, It will default to arbitrary
+deployment without zonal awareness.
+
+### Mon Failover
```

Comment:
The paragraph about the mon canary just seems like part of the main design, no need for the failover heading.
```suggestion
```